File: c/1004.c
#include <stdio.h>
#include <math.h>

int t;

void arrInput ( int* arr , int m ) {
	int *const arrEnd = arr + m;
	for (; arr < arrEnd; ++arr)
		scanf("%d", arr);
}

int main () {
	int x[4], c[3];
	
	int cnt, n;
	_Bool a, b;
	
	scanf("%d", &t);
	while (t--){
		arrInput (x, 4);
		cnt = 0;
		
		scanf("%d",&n);
		while (n--){
			arrInput (c, 3);
			a = (float) c[2] > sqrt( pow (x[0]-c[0], 2) + pow (x[1]-c[1],2));
			b = (float) c[2] > sqrt( pow (x[2]-c[0], 2) + pow (x[3]-c[1],2));
			
			if ( (!a && b)||(a && !b) )
				cnt++;
		}
		printf("%d\n", cnt);
	}
	return 0;
}

File: py/1004.py
from sys import stdin
from math import sqrt

t = int(stdin.readline())

for _ in range(t):
	x, y, z, w = map(int,stdin.readline().split())
	cnt = 0
	n = int(stdin.readline())
	for _ in range(n):
		cx, cy, r = map(int,stdin.readline().split())
		a = r > sqrt((x-cx)**2+(y-cy)**2)
		b = r > sqrt((z-cx)**2+(w-cy)**2)
		
		if (not a and b) or (a and not b):
			cnt += 1
			
	print(cnt)